service/src/loader.rs

```diff
@@ -42,31 +42,24 @@ impl Loader {
             let raw = entrypoint();
 
             Ok(Self {
-                service: Box::from_raw(raw),
+                service: Arc::new(Mutex::new(Box::from_raw(raw))),
                 so: lib,
             })
         }
     }
 
-    pub fn load(
-        self,
-        factory: &mut dyn Factory,
-        addr: SocketAddr,
-    ) -> Result<(ServeHandle, Library, ShutdownHandle), Error> {
-        let mut service = self.service;
-        let shutdown_handle = service.shutdown_handle();
+    pub fn load(mut self, factory: &mut dyn Factory, addr: SocketAddr) -> Result<Self, Error> {
+        if let Ok(mut svc) = self.service.lock() {
+            svc.build(factory)?
+        }
 
-        service.build(factory)?;
+        let service = Arc::clone(&self.service);
 
         // We cannot use spawn here since that blocks the api completely. We suspect this is because `bind` makes a blocking call,
         // however that does not completely makes sense as the blocking call is made on another runtime.
-        let handle = tokio::task::spawn_blocking(move || service.bind(addr));
-
-        // i want this
-        // this is just a mock
-        // let shutdown_handle = || {};
+        tokio::task::spawn_blocking(move || service.lock().unwrap().bind(addr));
```

Comment:
lock forever

Reply:
Lol, fun :sweat: Let's try not blocking...